src/components/ScaleLattice.vue

```diff
@@ -0,0 +1,340 @@
+    return combinations
+      .map(([a, b]) => {
+        return {
+          distance: monzoEuclideanDistance(
+            a.monzo.toIntegerMonzo().slice(1),
+            b.monzo.toIntegerMonzo().slice(1)
+          ),
+          pair: [a.name, b.name],
+        };
+      })
+      .filter(({ distance }) => distance === 1);
+  } catch {
+    return [];
+  }
+});
+
+const svgElement = ref<HTMLElement | null>(null);
+
+const viewBox = computed(() => {
+  const defaultViewBox = { minX: 0, minY: 0, maxX: 0, maxY: 0 };
+  if (svgElement.value) {
+    return [
+      ...svgElement.value.querySelectorAll("text"),
+      ...svgElement.value.querySelectorAll("circle"),
+    ].reduce(({ minX, maxX, minY, maxY }, el) => {
+      const { x, y, width, height } = el.getBBox();
+      return {
+        minX: minX < x - width / 2 ? minX : x - width / 2,
+        minY: minY < y - height / 2 ? minY : y - height / 2,
+        maxX: maxX > x + width ? maxX : x + width,
+        maxY: maxY > y + height ? maxY : y + height,
+      };
+    }, defaultViewBox);
+  }
+  return defaultViewBox;
+});
+
+function isNodeErrorMessage(nodeValue: any): nodeValue is string {
+  return nodeValue instanceof String;
+}
+
+const error = computed(() => {
+  if (!canLatticePrimes.value) {
+    return "Can only make lattice of JI scales up to 23-limit";
+  }
+  if (isNodeErrorMessage(nodes.value)) {
+    return (
+      nodes.value ||
+      "There was an unknown error while making this lattice, please try again or with another scale."
+    );
+  }
+  return null;
+});
+</script>
+
+<template>
+  <div ref="container" class="container">
+    <div v-if="error">
+      <p>{{ error }}</p>
+    </div>
+    <div v-else>
+      <svg
+        v-if="Array.isArray(nodes)"
+        ref="svgElement"
+        id="lattice"
+        class="lattice"
+        xmlns="http://www.w3.org/2000/svg"
+        preserveAspectRatio="xMidYMid meet"
+        style="background-color: transparent"
+        :viewBox="
+          [
+            viewBox.minX,
+            viewBox.minY,
+            viewBox.maxX - viewBox.minX,
+            viewBox.maxY - viewBox.minY,
+          ].join(' ')
+        "
+      >
+        <line
+          v-for="edge of edges"
+          class="edge"
+          :key="edge.pair.join()"
+          :x1="nodesByLabel[edge.pair[0]].x"
+          :x2="nodesByLabel[edge.pair[1]].x"
+          :y1="nodesByLabel[edge.pair[0]].y"
+          :y2="nodesByLabel[edge.pair[1]].y"
+        />
+        <circle
+          v-for="node of nodes"
+          :key="node.label"
+          :cx="node.x"
+          :cy="node.y"
+          r="2.5"
+          :class="`node   ${
+            props.heldScaleDegrees.has(node.index) ? 'heldDegree' : ''
+          }`"
+        />
+        <!-- Separating the iteration of the labels so that circles don't overlap any text -->
+        <text
+          v-for="node of nodes"
+          :key="node.label + 'text'"
```

Comment:
See issue about duplicity above. Keys only need to be unique within the `v-for` so `+ 'text'` is superfluous here.